### PR TITLE
Implement login

### PR DESCRIPTION
**Pull request by andymond**
Description of Changes:
-
* Add sessions controller w/ login & logout routes
* Remove password_confirmation validation, prevented any users from being created

### Tests Added:
* User can log in, unregistered user can't login
* User can log out regardless of account status

#### Reference Related Issue(s) or Spec Requirement:
- #.

#### Checklist:
- [x] Code follows code style of this project
- [x] Tests added to cover changes
- [x] All new and existing tests passing
#

#### Questions / Additional Notes:

@andymond @lnchambers

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  def new
+  end
+
+  def create
+    if user = User.find_by(email: params[:email])
+      session[:user_id] = user.id
+      flash[:notice] = "Welcome #{user.name}!"
+      redirect_to dashboard_path
+    else
+      flash[:error] = "Miss! Invalid Credentials"
```
Comment (lnchambers) [inline app/controllers/sessions_controller.rb]: I love the use of battleship puns

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      redirect_to login_path
+    end
+  end
+
+  def destroy
+    session.destroy
+    @current_user = nil
+    flash[:notice] = "Successfully logged out"
```
Comment (lnchambers) [inline app/controllers/sessions_controller.rb]: What if we had "You sunk your own battleship!"?

File: app/models/user.rb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 class User < ApplicationRecord
   has_secure_password
-  validates_presence_of :password_confirmation, :password, :email, :name
+  validates_presence_of :password, :email, :name
```
Comment (lnchambers) [inline app/models/user.rb]: Does the confirmation still work?

File: app/views/user_mailer/activation_invite.html.erb [hunk 4]
```diff
@@ -7,5 +7,6 @@
     <h2>Welcome to BattleShift!</h2>
     <p>Thanks for signing up <%= @user.name %>!</p>
     <%= link_to "Click here to activate your account", @url %>
+    <p>Your api key is: <%= @user.id %>, please keep it secret.</p>
```
Comment (lnchambers) [inline app/views/user_mailer/activation_invite.html.erb]: Can we add "Forward this email to ten friends or hailstones the size of destroyers will rain down on your car" to the end?

Comment (andymond) [reply app/views/user_mailer/activation_invite.html.erb]: Absolutely I meant to do that

File: app/models/user.rb (hunk 3 above)
Comment (andymond) [reply app/models/user.rb]: Yep, there is still a validates_confirmation_of :password on the model

**Pull request merged**